platforms/software/dependency-management/src/main/java/org/gradle/api/internal/artifacts/configurations/DefaultConfiguration.java

```diff
@@ -194,6 +197,7 @@ public abstract class DefaultConfiguration extends AbstractFileCollection implem
     private ResolutionStrategyInternal resolutionStrategy;
     private final FileCollectionFactory fileCollectionFactory;
     private final ResolveExceptionContextualizer exceptionContextualizer;
+    private final RepositoriesSupplier repositoriesSupplier;
```

Comment:
🤔 Why does this type even exist?  Is it just a typealias at this point?

Have you considered expanding it to a full class and moving `shouldSkipRepository` and `hasNonRequiredAttribute` there, so we can just ask it for repositories?  It wouldn't be a `java.util.function.Supplier`, as the new `get` equivalent would have to accept a `Configuration` to look for acceptable repos for, but it would be nice to avoid expanding `DefaultConfiguration` further.

Reply:
It is mostly a type alias. The purpose of the alias is so it can be service-injected, since you can't inject generic types.

The way I understand the `RepositoriesSupplier` is that it provides all _possible_ repositories for a project/script/etc. It combines the `dependencyResolutionManagement` repos with those defined in the project. 

It should not be concerned with the consumers of those repositories. The high-level goal here is to _reduce_ the number of places that require a `Configuration` (or `ResolveContext`), so having the repositories supplier _accept_ a `Configuration` would go against that. 

I hear what you mean about avoiding expanding the `DefaultConfiguration` type. My motivation for this change is to separate the `ConfigurationResolver` from a `Configuration`. This means working towards having a `ConfigurationResolver` no longer accept a `ResolveContext` (a subset of Configuration), but instead accept a context-free set of parameters required for _some_ resolution. Effectively, this means converting the `ConfigurationResolver` to a `AnythingResolver`, allowing us to resolve things that aren't Configurations.

To me, this means that if we want to avoid adding more logic to `DefaultConfiguration` we need to create some kind of middle-man between the existing logic in `DefaultConfigurationResolver` and `DefaultConfiguration`.

Reply:
Got it.  That makes sense.  I think the change to Move control of filtering configuration respositories to the resolver is also an improvement.